frontend/src/app/edit-claim/edit-claim.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute, ParamMap } from '@angular/router';
+import { MiddlewareService } from '../middleware/middleware.service';
+import { switchMap } from 'rxjs/operators';
+import { FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'app-edit-claim',
+  templateUrl: './edit-claim.component.html',
+  styleUrls: ['./edit-claim.component.scss']
+})
+export class EditClaimComponent implements OnInit {
+  private claim: Protos.Claim
+
+  files: FileList;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private service: MiddlewareService
+  ) { }
+
+  ngOnInit() {
+    const observable = this.route.paramMap.pipe(
+      switchMap((params: ParamMap) =>
+        this.service.getClaim(params.get('vin')))
```

Comment:
This is wrong.

You do understand what url get params are, correct? This implies that the URL has the following structure.

```
/claims?vin=<vin-number>
```

This makes me so sad.

Reply:
Angular is such a dumpster fire. Just looked into the param thing and this _might_ be correct, however, it's still not setting `claim` correctly in the html file. The claim is being returned by the middleware.